### PR TITLE
Bump Docker GX OSS version to `0.18.7`

### PR DESCRIPTION
**Pull request by Kilo59**
Bump Docker GX OSS version to `0.18.7`

Update README

Added instructions for updating Dockerfile dependencies (the `poetry.lock` file)

### CODE REVIEW

File: poetry.lock [hunk 1]
```diff
@@ -752,13 +752,13 @@ files = [
 
 [[package]]
 name = "great-expectations"
-version = "0.18.6"
+version = "0.18.7"
```
Comment (Kilo59) [inline poetry.lock:755]: ☝️

File: README.md [hunk 2]
```diff
@@ -31,14 +31,16 @@ Also available as an invoke task.
...
-#### Release to PyPI
+#### Release to PyPI and Docker
 
 To release a new version to PyPI the version must be incremented.
 New versions are automatically published to PyPI when merging to `main`.
 ```console
 invoke version-bump
 ```
 
+A new docker tag will also be generated and pushed to [dockerhub](https://hub.docker.com/r/greatexpectations/agent).
```
Comment (JennyTee) [inline README.md:42]:
```suggestion
A new docker tag will also be generated and pushed to [Docker Hub](https://hub.docker.com/r/greatexpectations/agent).
```

**Review by JennyTee — APPROVED** [review]
✨

**Review by billdirks — APPROVED** [review]
LGTM, thanks!

**Pull request merged**